exercises/practice/square-root/square-root.jl

```diff
@@ -0,0 +1,3 @@
+function square_root(n::Int64; tol)
```

Comment:
Do we need to provide a tolerance parameter? The built-in sqrt doesn't use one.

If we do provide a tolerance parameter then I think we should specify if it is an absolute tolerance (atol) or relative tolerance (rtol) to match base Julia's naming patterns.

Reply:
Here's an implementation that will converge on the exact answers for integer square numbers:

```julia
function mysqrt2(S)
    x = S
    while (S/x + x) / 2 ≠ x
        x = (S/x + x) / 2
    end
    return x
end


findall(sqrt(n^2) ≠ mysqrt2(n^2) for n in 1:1000)
```

Reply:
I thought having a keyword argument would be a nice practice that is usually not done in these exercises, but that's not a must.

Reply:
Cool, well I think it's better to not have it in this exercise. Thanks for removing it!